torch/csrc/jit/codegen/cuda/fusion.h

```diff
@@ -191,6 +197,12 @@ class TORCH_CUDA_CU_API Fusion final {
   bool hasInput(const Val* val) const;
   bool hasOutput(const Val* val) const;
 
+ protected:
+  friend SegmentCandidateFinder;
```

Comment:
part of finding an alternative to Fusion::copy() is removing these friends too

Reply:
Will be addressed in PRs related to issue #643